pydy/bodies.py

```diff
@@ -0,0 +1,168 @@
+    If nothing is passed, then by default, a RigidBody is created. User can
+    also pass masscenter, mass, frame and inertia to change the arguments
+    passed to RigidBody.
+    To create a Particle only mass and masscenter needs to be passed. User can
+    also pass a frame which will be attached to the Particle.
+
+    In short, it creates a Rigidbody if inertia is passed or nothing is passed.
+    and a Particle if inertia is not passed and things like masscenter and mass
+    are.
+
+    Parameters
+    ---------
+    name: string
+        defines the name of the body. It is used as the base for defining body
+        specific properties.
+    masscenter : Point (optional)
+        The point which represents the center of mass of the rigid body.
+    frame : ReferenceFrame (optional)
+        The ReferenceFrame which the rigid body is fixed in.
+    mass : Sympifyable (optional)
+        The body's mass.
+    inertia : (Dyadic, Point) (optional)
+        The body's inertia about a point; stored in a tuple as shown above.
+
+    Example:
+    --------
+    1. Default behaviour. It creates a RigidBody after defining mass,
+     masscenter, frame and inertia.
+
+    >>> from pydy.bodies import Body
+    >>> body = Body('name_of_body')
+
+    2. Passing attributes of Rigidbody. All the arguments needed to create a
+     RigidBody can be passed while creating a Body too.
+
+    >>> from sympy import Symbol
+    >>> from sympy.physics.mechanics import ReferenceFrame, Point
+    >>> from pydy.bodies import Body
+    >>> mass = Symbol('mass')
+    >>> masscenter = Point('masscenter')
+    >>> frame = ReferenceFrame('frame')
+    >>> I = outer (A.x, A.x)
+    >>> inertia_tuple = (I, P)
+    >>> body = Body('name_of_body', masscenter, mass, frame, inertia_tuple)
+
+    3. Creating a Particle. If masscenter and mass are passed, and inertia is
+     not then a Particle is created.
+
+    >>> from sympy import Symbol
+    >>> from sympy import Point
+    >>> from pydy.bodies import Body
+    >>> mass = Symbol('mass')
+    >>> masscenter = Symbol('masscenter')
+    >>> body = Body('name_of_body', masscenter, mass)
+
+    Similarly, A frame can also be passed while creating a Particle.
+
+    """
+    def __init__(self, name, masscenter=None, mass=None, frame=None,
+                 inertia=None):
+
+        self.name = name
+        self.parent = None
+        self.child = None
+        self.force_list = []
+        self.body = None  # TODO assign a better name.
+        self._coordinates = []
+        self._speeds = []
+        self._counter = 0
+
+        if masscenter is None:
+            self._masscenter = Point(self.name + '_masscenter')
+        else:
+            self._masscenter = masscenter
+
+        if mass is None:
+            self._mass = Symbol(self.name + '_mass')
+        else:
+            self._mass = mass
+
+        if frame is None:
+            self._frame = ReferenceFrame(self.name + '_frame')
```

Comment:
Instead of creating a ReferenceFrame and Point, etc, shouldn't a RigidBody or Particle be created in the class. That way you will  remove redundancy and can get all of the other methods and attributes on RigidBody and Particle for free.

I'm still not sold on this body class as is.

I'm wondering if it should somehow be a super class of rigidbody and particle...